prereise/gather/demanddata/bldg_electrification/load_projection.py

```diff
@@ -0,0 +1,674 @@
+            + self.stats["frac_resist_com_heat"] * self.floor_area_m2["com"]
+        )
+        self.resist_heat_frac = self.resist_heat_area_m2 / (
+            self.floor_area_m2["res"] + self.floor_area_m2["com"]
+        )
+        self.elec_cool_m2 = (
+            self.stats["frac_elec_res_cool"] * self.floor_area_m2["res"]
+            + self.stats["frac_elec_com_cool"] * self.floor_area_m2["com"]
+        )
+        return
+
+    def floor_area_growth(self, other):
+        # compound floor area growth
+        return (self.floor_area_m2["res"] + self.floor_area_m2["com"]) / (
+            other.floor_area_m2["res"] + other.floor_area_m2["com"]
+        )
+
+    def floor_area_growth_type(self, other, clas):
+        # compound floor area growth by building type
+        return self.floor_area_m2[clas] / other.floor_area_m2[clas]
+
+    def frac_hp_growth(self, other):
+        # floor area growth ratio that use hp as main heating appliance
+        return self.hp_heat_area_m2 / other.hp_heat_area_m2
+
+    def frac_resist_growth(self, other):
+        # floor area growth ratio that use resistance heat as main heating source
+        return self.resist_heat_area_m2 / other.resist_heat_area_m2
+
+    def frac_cool_growth(self, other):
+        # floor area growth ratio that have electric air conditioning
+        return self.elec_cool_m2 / other.elec_cool_m2
+
+    def frac_htg_ff2hp(self, other, clas):
+        # fraction of floor area electrified for heating
+        return other.stats[f"frac_ff_{clas}_heat"] - self.stats[f"frac_ff_{clas}_heat"]
+
+    def frac_dhw_ff2hp(self, other, clas):
+        # fraction of floor area electrified for dhw
+        return other.stats[f"frac_ff_dhw_{clas}"] - self.stats[f"frac_ff_dhw_{clas}"]
+
+    def frac_cook_ff2hp(self, other, clas):
+        # fraction of floor area electrified for cooking
+        cook_other = "cook" if clas == "com" else "other"
+        return (
+            other.stats[f"frac_ff_{cook_other}_{clas}"]
+            - self.stats[f"frac_ff_{cook_other}_{clas}"]
+        )
+
+    def frac_cooling_eff_change(self, other):
+        # account for future cooling efficiency improve
+        return self.cool_energy_intensity / other.cool_energy_intensity
+
+    def compare_hp_heat_type(self, other):
+        # return True if the heat pump type for projection scenario is the same as that of base scenario
+        return self.hp_type_heat == other.hp_type_heat
+
+
+def temp_to_energy(temp_series, hourly_fits_df, db_wb_fit, base_scen, hp_heat_cop):
+    """Compute baseload, heating, and cooling electricity for a certain hour of year under model base year scenario
+
+    :param pandas.Series temp_series: data for the given hour.
+    :param pandas.DataFrame hourly_fits_df: hourly and week/weekend breakpoints and
+        coefficients for electricity use equations.
+    :param class scenarios base_scen: reference scenario class
+    :return: (*list*) -- [baseload, heating, cooling]
```

Comment:
Doc strings need to be consistent with the parameters.

Reply:
done.